apiserver/objects.go

```diff
@@ -216,12 +216,6 @@ func (h *objectsCharmHTTPHandler) processPut(r *http.Request, st *state.State, a
 	}
 
 	source := charm.Schema(schema)
-	if source == charm.Local {
-		curl, err = st.PrepareLocalCharmUpload(curl.String())
-		if err != nil {
-			return nil, errors.Capture(err)
-		}
-	}
```

Comment:
I think we can now drop `st` from this handler entirely. It's only used to check we don't upload ch charms unless we're migrating, a check which I think @SimonRichardson has pushed into the service?

Reply:
I haven't found any method to know if a charm is being migrated in the application domain, maybe it's part of an unmerged PR? 
I can come back and remove it once we have this on the domain.